tljh/jupyterhub_config.py

```diff
@@ -13,12 +13,16 @@
 from tljh.yaml import yaml
 from jupyterhub_traefik_proxy import TraefikTomlProxy
 
+from traitlets import Dict
+
 class UserCreatingSpawner(SystemdSpawner):
     """
     SystemdSpawner with user creation on spawn.
 
     FIXME: Remove this somehow?
     """
+    user_groups = Dict(config=True)
```

Comment:
Try this:

```suggestion
    user_groups = Dict(key_trait=Unicode(), value_trait=List(Unicode()), config=True)
```

This means that the keys will be validated as text (group names) and the values as lists of text (lists of group names). So if someone inputs by accident e.g.:

```python
c.UserCreatingSpawner.user_groups = {'group': 'user'}
```

the error should be caught at config time. The code as it is, would treat this the same as `{'group': ['u', 's', 'e', 'r']}` since a string is an iterable just like a list.

Reply:
Thanks for the suggestion @minrk, this is really nice. After 3 commits I finally managed to get it right :grimacing: